src/main/kotlin/study/domain/RaceGame.kt

```diff
@@ -0,0 +1,32 @@
+package study.domain
+
+import kotlin.random.Random
+
+data class RaceGame internal constructor(
+    private val raceCars: List<RaceCar>,
+    private val totalRound: Int,
+    private val moveStrategy: () -> Boolean = fun(): Boolean = Random.nextInt(10) >= 4,
+) {
+    fun play(): RaceResult {
+        val raceResult = RaceResult.empty()
+        repeat(totalRound) { round ->
+            raceCars.forEach { car ->
+                if (moveStrategy.invoke()) {
```

Comment:
```suggestion
                if (moveStrategy()) {
```
람다를 호출하는 경우 invoke 는 생략 할수 있어요 😄 
https://wooooooak.github.io/kotlin/2019/03/21/kotlin_invoke/